pack/nodejs/helm.ts

```diff
@@ -0,0 +1,715 @@
+import { execSync } from "child_process";
+import * as fs from "fs";
+
+import * as k8s from "./index";
+import * as pulumi from "@pulumi/pulumi";
+import * as shell from "shell-quote";
+import * as tmp from "tmp";
+import * as yaml from "js-yaml";
+
+export namespace v2 {
+    export interface ChartOpts {
+        repo: string;
+        chart: string;
+        version: string;
+
+        values?: any;
+        transformations?: transform.Transform[];
+        fetchOpts?: FetchOpts;
+    }
+
+    // A CustomResource representing a Helm Chart.
+    //
+    // `Chart` does not use Tiller. The Chart specified is copied and expanded locally; any values
+    // that would be retrieved in-cluster would be assigned fake values, and none of Tiller's
+    // server-side validity testing is executed.
+    //
+    // Important caveat: Charts often assign hard-coded names for objects; since Pulumi is not able
+    // to assign such names, any time a resource must be replaced, it will be marked as
+    // delete-before-replace. For example, in a normal Pulumi program, a change in a ConfigMap's
+    // would result in: (1) a new ConfigMap being created, (2) all Deployments referencing that
+    // ConfigMap being updated to refer to the new ConfigMap, and (3) deleting the old ConfigMap
+    // only after all the rollouts of those Deployments have succeeded. In contrast, since the
+    // chart asks for a specific name for this ConfigMap, it must be deleted before being replaced
+    // with a ConfigMap of the same name.
+    export class Chart extends pulumi.ComponentResource {
+        public readonly resources: pulumi.CustomResource[];
+
+        constructor(releaseName: string, config: ChartOpts, opts?: pulumi.ResourceOptions) {
+            super("kubernetes:helm.sh/v2:Chart", releaseName, config, opts);
```

Comment:
Why `helm.sh` here?

Reply:
Because Kubernetes resources are organized by group/version/kind. The group is `helm.sh`, the API version is `v2`, and the kind is `Chart`. By convention `kubernetes:v2:Chart` refers to `core/v2/Chart`, which is confusing.

Reply:
Question really was why is it `helm.sh` here but `helm` in the module name.  (The latter of course can't have a `.`, but assuming this `helm.sh` name isn't derived from an actual Kubernetes object, why not just `helm`?

Reply:
Because groups tend to be identified by the domain that the organization that "owns" that group owns. For example, `pulumi.com` vs `pulumi`, `ksonnet.io` vs `ksonnet`, and so on.

Reply:
I think we've agreed to keep this as is.